pebble/common.py

```diff
@@ -124,16 +124,18 @@ def rebuild_exception(exception, traceback):
     return exception
 
 
-def launch_thread(function, *args, **kwargs):
-    thread = Thread(target=function, args=args, kwargs=kwargs)
```

Comment:
This is not where you pop the `name` keyword. You can check the `concurrent.process` decorator to see how it is done.

Reply:
I think I still have to pop the name kwarg from the `kwargs` here since these are not the real end used function kwargs. I mean that I need to have the name at this point but I can't pass it as an argument of `launch_thread` since it already has `**kwargs`. Sorry, if it's not clear but it's not really easy to explain.